x/bank/types/zerocopy/msgsend.go

```diff
@@ -0,0 +1,72 @@
+package zerocopy
+
+const (
+	msgSendFromAddressOffset = 0
+	msgSendToAddressOffset   = msgSendFromAddressOffset + 4
+	msgSendCoinsOffset       = msgSendToAddressOffset + 4
+	msgSendSize              = msgSendCoinsOffset + 4
+)
+
+type MsgSend struct {
+	ctx *BufferContext
+}
+
+func (m *MsgSend) FromAddress() string {
+	if m.ctx == nil {
+		return ""
+	}
+	return m.ctx.ResolvePointer(msgSendFromAddressOffset).ReadString()
+}
+
+func (m *MsgSend) ToAddress() string {
+	if m.ctx == nil {
+		return ""
+	}
+	return m.ctx.ResolvePointer(msgSendToAddressOffset).ReadString()
+}
+
+func (m *MsgSend) init() {
+	if m.ctx == nil {
+		_, m.ctx = (&Buffer{}).Alloc(msgSendSize)
+	}
+}
+
+func (m *MsgSend) SetFromAddress(x string) *MsgSend {
+	m.init()
+	m.ctx.SetString(msgSendFromAddressOffset, x)
+	return m
+}
+
+func (m *MsgSend) SetToAddress(x string) *MsgSend {
+	m.init()
+	m.ctx.SetString(msgSendToAddressOffset, x)
+	return m
+}
+
+func (m *MsgSend) Coins() Array[*Coin] {
+	m.init()
+	return ReadArray[Coin](m.ctx, msgSendCoinsOffset)
+}
+
+func (m *MsgSend) InitCoins(size int) Array[*Coin] {
```

Comment:
Arrays must be allocated as fixed size with this approach and can't be appended to (at least without re-organizing the memory buffer). To change the size the array needs to be re-initialized and re-populated